core/templates/pages/exploration-player-page/services/diagnostic-test-player-engine.service.ts

```diff
@@ -0,0 +1,291 @@
+
+        stateCard = this.createCard();
+        focusLabel = this.focusLabel;
+
+        successCallback(
+          stateCard, refreshInteraction, feedbackHtml,
+          feedbackAudioTranslations, refresherExplorationId,
+          missingPrerequisiteSkillId, remainOnCurrentCard,
+          taggedSkillMisconceptionId, wasOldStateInitial, isFirstHit,
+          isFinalQuestion, focusLabel
+        );
+        return answerIsCorrect;
+      });
+    } else {
+      stateCard = this.createCard();
+      focusLabel = this.focusLabel;
+
+      successCallback(
+        stateCard, refreshInteraction, feedbackHtml,
+        feedbackAudioTranslations, refresherExplorationId,
+        missingPrerequisiteSkillId, remainOnCurrentCard,
+        taggedSkillMisconceptionId, wasOldStateInitial, isFirstHit,
+        isFinalQuestion, focusLabel
+      );
+      return answerIsCorrect;
+    }
+  }
+
+  getLanguageCode(): string {
+    return (
+      this.diagnosticTestCurrentTopicStatusModel.getNextQuestion(
+        this.currentSkillId).getLanguageCode());
+  }
+
+  recordNewCardAdded(): void {
+    this.contextService.setCustomEntityContext(
+      AppConstants.ENTITY_TYPE.QUESTION, this.currentQuestion.getId() as string
+    );
+  }
+
+  makeQuestion(
+      newState: State, envs: Record<string, string>[]): string {
+    return this.expressionInterpolationService.processHtml(
+      newState.content.html, envs);
+  }
+
+  createCard(): StateCard {
+    if (!this.diagnosticTestCurrentTopicStatusModel.isLifelineConsumeed()) {
+      this.currentSkillId = (
+        this.diagnosticTestCurrentTopicStatusModel.getNextSkill());
+    }
+    this.currentQuestion = (
+      this.diagnosticTestCurrentTopicStatusModel.getNextQuestion(
+        this.currentSkillId));
+    const stateData: State = this.currentQuestion.getStateData();
+    const questionHtml: string = this.makeQuestion(stateData, []);
+
+    if (questionHtml === '') {
+      this.alertsService.addWarning('Question name should not be empty.');
+    }
+
+    this.focusLabel = this.focusManagerService.generateFocusLabel();
+    const interaction = stateData.interaction;
+    const interactionId = interaction.id;
+    let interactionHtml: string = '';
+
+    if (interactionId) {
+      interactionHtml = (
+        this.explorationHtmlFormatterService.getInteractionHtml(
+          interactionId, interaction.customizationArgs,
+          true, this.focusLabel, null)
+      );
+    }
+
+    return StateCard.createNewCard(
+      stateData.name as string, questionHtml, interactionHtml as string,
+      interaction, stateData.recordedVoiceovers, stateData.writtenTranslations,
+      stateData.content.contentId as string,
+      this.audioTranslationLanguageService
+    );
+  }
+
+  static get MAX_ALLOWED_QUESTIONS_IN_THE_DIAGNOSTIC_TEST(): number {
+    return AppConstants.MAX_ALLOWED_QUESTIONS_IN_THE_DIAGNOSTIC_TEST;
+  }
+
+  isDiagnosticTestFinished(): boolean {
+    const lengthOfEligibleTopicIds = (
```

Comment:
I thought we moved away from the wording "eligible" because it's not really understandable to devs who aren't that familiar with the constructs here?

Reply:
Rephrased to pendingTopicIdsToTest.